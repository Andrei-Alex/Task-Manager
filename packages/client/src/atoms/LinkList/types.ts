import { NavElement } from "../../../../libs/sharedTypes";

export interface ILinkList {
  listElements: NavElement[];
}
