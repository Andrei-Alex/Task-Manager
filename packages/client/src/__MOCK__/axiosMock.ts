/**
 * Mock Axios
 * @return {Promise}
 */
export const axiosMock = () => {
  return {
    get: jest.fn(() => Promise.resolve({ data: {} })),
  };
};
