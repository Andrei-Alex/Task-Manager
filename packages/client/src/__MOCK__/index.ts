export { routerMock } from './router/routerMock';
export { axiosMock } from './axios/axiosMock';
