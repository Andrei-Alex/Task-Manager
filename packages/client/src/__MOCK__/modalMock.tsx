import { LayoutContext } from "@/providers";
import { useState } from "react";
import { Modal } from "@/atoms";

export const WithMockLayoutContext = () => {
  const [isVisible, setVisible] = useState(true);
  const [isBurgerOpen, setIsBurgerOpen] = useState(false);
  const isMobile = true;
  return (
    <LayoutContext.Provider value={[isMobile, isBurgerOpen, setIsBurgerOpen]}>
      <Modal
        isVisible={isVisible}
        visibilityHandler={() => setVisible(false)}
        customContainerStyles={{
          width: "250px",
          height: "300px",
        }}
        footerElements={"Footer"}
      >
        Body
      </Modal>
    </LayoutContext.Provider>
  );
};
