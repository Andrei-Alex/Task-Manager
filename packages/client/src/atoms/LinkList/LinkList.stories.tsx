import type { Meta, StoryObj } from "@storybook/react";
import React from "react";
import {LinkList} from "./LinkList";

const meta: Meta<typeof LinkList> = {
    component: LinkList,
    title: "Atoms/LinkList",
    tags: ["autodocs"],
    argTypes: {
    },

};
export default meta;
type Story = StoryObj<typeof LinkList>;

export const Default: Story = {
    args: {
        listElements: [
            {id: 'Home',name: 'Home', navigateTo:'home'},
            {id: 'Board',name: 'Board', navigateTo:'board'},
            {id: 'Login',name: 'Login', navigateTo:'login'}
        ]
    },
};

