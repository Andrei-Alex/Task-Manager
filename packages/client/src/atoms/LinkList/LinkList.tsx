import { styles, ILinkList } from ".";
import Link from "next/link";
import React from "react";

const LinkList: React.FC<ILinkList> = ({ listElements }) => {
  return (
    <ul>
      {listElements.map((navElement: any) => (
        <li key={navElement.id}>
          <Link href={navElement.navigateTo}>{navElement.name}</Link>
        </li>
      ))}
    </ul>
  );
};
function PropsAreEqual(prevProps: ILinkList, nextProps: ILinkList) {
  return JSON.stringify(prevProps.listElements) === JSON.stringify(nextProps.listElements)
}
export default React.memo(LinkList, PropsAreEqual);
