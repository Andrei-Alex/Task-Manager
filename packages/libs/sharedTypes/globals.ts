import React from "react";

export type children = React.ReactNode | React.ReactNode[];
export type Obj = { [key: string]: string | number | boolean | Function };
export interface ILayouts {
  children: children;
}
