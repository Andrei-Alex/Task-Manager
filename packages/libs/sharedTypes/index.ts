export * from "./globals";
export * from "./navigations";
