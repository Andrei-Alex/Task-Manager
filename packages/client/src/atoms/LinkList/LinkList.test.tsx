import React from "react";
import {render, fireEvent } from "@testing-library/react";
import renderer from "react-test-renderer";
import { screen } from "@testing-library/dom";
import LinkList from "./LinkList";
import {RouterContext} from "next/dist/shared/lib/router-context";
import {routerMock} from "@/__MOCK__";
jest.mock('next/router');
import {NavElement} from "../../../../libs/sharedTypes";

const mockedLinks: NavElement[]= [{
    id:'link-1',
    name: 'Home',
    navigateTo: 'PageOne'
    },
    {
        id:'link-2',
        name: 'Board',
        navigateTo: 'PageTwo'
    }]
describe("NavLinks", () => {
    it("renders correctly", () => {
        const tree = renderer.create(<LinkList listElements={mockedLinks}/>).toJSON();
        expect(tree).toMatchSnapshot();
    });
    it("Should Display links", () => {
        render(<LinkList listElements={mockedLinks}/>);
        const firstTitle = screen.getByText("Home");
        expect(firstTitle).toBeInTheDocument();
        expect(firstTitle).toHaveAttribute('href', 'PageOne');
        const secondTitle = screen.getByText("Board")
        expect(secondTitle).toBeInTheDocument();
        expect(secondTitle).toHaveAttribute('href', 'PageTwo');
    });
    it('calls the push function when clicked', () => {
        const mockHandleLinkClick = jest.fn();
        render(
        <RouterContext.Provider value={routerMock({push: (url) => mockHandleLinkClick(url)})}>
            <LinkList listElements={mockedLinks} />
        </RouterContext.Provider>
        );
        const linkElement = screen.getByText("Home");
        fireEvent.click(linkElement);
        expect(mockHandleLinkClick).toHaveBeenCalledTimes(1);
        expect(mockHandleLinkClick).toHaveBeenCalledWith('/PageOne');
    });
});


