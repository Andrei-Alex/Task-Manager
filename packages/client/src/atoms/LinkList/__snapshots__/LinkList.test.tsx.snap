// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavLinks renders correctly 1`] = `
<ul
  className="list"
>
  <li>
    <a
      className="link"
      href="PageOne"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Main Title
    </a>
  </li>
  <li>
    <a
      className="link"
      href="PageTwo"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      Second Title
    </a>
  </li>
</ul>
`;
